modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -5,6 +5,132 @@
  * Hook implementations for Facebook Instant Articles Display module.
  */
 
+/**
+ * Load all Field module hooks for Facebook Instant Articles Display module.
+ */
+require_once __DIR__ . '/includes/field.inc';
+
+/**
+ * Implements hook_entity_load().
+ */
+function fb_instant_articles_display_entity_load($entities, $type) {
+  $enabled_entity_types = fb_instant_articles_display_get_article_entity_types();
+  foreach ($entities as $entity) {
+    // Only proceed if the entity is of the configured entity type and bundle.
+    list($id, $vid, $bundle) = entity_extract_ids($type, $entity);
+    if (!in_array($type, array_keys($enabled_entity_types))
+      || !in_array($bundle, array_keys($enabled_entity_types[$type]))) {
+      return FALSE;
+    }
+
+    // We add Entity-specific information to the ArticleWrapper object, and
+    // attach it to the Entity for use in later hooks. Note that we don't store
+    // the InstantArticle object on the node directly, because our Singleton
+    // wrapper + Drupal hook class pattern relies on invoking
+    // hook_fb_instant_articles_article_alter() in ArticleWrapper->getArticle(),
+    // to allow other modules to alter the InstantArticle object whenever it is
+    // requested. If we request that now, it will not have the Entity-specific
+    // information on it this module needs for later altering.
+    $context = array(
+      'entity_type' => $type,
+      'entity' => $entity,
+    );
+    $wrapper = \Drupal\fb_instant_articles\ArticleWrapper::create($context);
```

Comment:
@m4olivei Excellend catch. This is why I make comments like this 😄  Now that we've moved the alter hook to `ArticleWrapper::__construct()` the reason I called `EntityPropertyMapper::create()` in `hook_entity_load()` is now moot, and even why we store `ArticleWrapper` instead of `InstantArticle` on the Entity is now not needed. In 089e621 I changed strategy to storing the `InstantArticle` object directly onto the Entity. I also now call `EntityPropertyMapper` from `hook_fb_instant_articles_article_alter()` instead of `hook_entity_load()` (and modified our `hook_module_implements_alter()` so our `hook_fb_instant_articles_article_alter()` run first, giving modules the ability to alter our default property mappings in `EntityPropertyMapper::map()`). 

Note that none of this allows us an alter hook to modify the `InstantArticles` object after all the *field* mappings are finished (I can see no way to allow that, and also follow this architecture, so that we get the value from the Display module for API usage)… but I guess a custom module could implement their own `hook_field_formatter_view()`, ensure it runs after the Display module implementation, and change the default mappings from `EntityFieldMapper::map()` that way.